src/app/interfaces/trace.ts

```diff
@@ -0,0 +1,27 @@
+export default interface Trace {
+    traceId: string
+    projectId : string
+    projectName: string
+    boardId: string
+    boardName: string
+    agentUserId: string
+    agentUserName: string
+    commentId: string
+    commentText: string
+    postId: string
+    postTitle: string
+    postMessage: string 
+    postTitleOrMessageModifiedCounter: number
+    clientTimestamp: number
+    serverTimestamp: number
+    commentModifiedTextCounter: number
+    postModifiedUpvote: number
+    postTagNameAdded: string[]
+    postTagNameRemoved: string
+    postModifiedLocationX: number | null
+    postModifiedLocationY: number | null
+    postDeleted: number
+    bucketId: string
+    bucketName: string
+    postRead: number 
```

Comment:
Won't that imply the Trace collection will have different set of fields for each document?

Reply:
Maybe something like 
```
Trace {
    serverTimestamp
    clientTimestamp
    ... any other mandatory fields

    eventType: EventType
    event: Event
}

interface Event {
   ... any common fields between all events
}

LikeEvent implements Event {
    likeAmount 
    .. any other trace data for a like event
}
```

Reply:
With the above example, no it wont

Reply:
@Ammar-T I honestly still don't get it..does it mean we will have a different collection called Event then?

Reply:
No event isn't it's own collection it's just a data structure used inside trace which is a collection

Reply:
So the trace contains another collection?

Reply:
@Ammar-T if we refactor trace to use eventType and event we would run into issues during the csv export. Since events have different fields, trace object would have different schemas (after the event object is flattened). In that case we would have to use a different library. [Json2csv ](https://www.npmjs.com/package/json-2-csv) supports different schemas and default values for fields. If you're good with this i'll start refactoring the code.

Reply:
@l3n0ire Oh I see what Steven meant now, do you think we should store the event object as a string then so that event doesn't have different fields? The string would just be created by doing `JSON.stringify(event)`. 

You can go with whatever seems best, I just wanted to throw this out there as well!

Reply:
Yeah thats pretty hacky lol. I'll try with eventType and event and json2csv and see how far I get.